cfme/tests/cloud_infra_common/test_tag_objects.py

```diff
@@ -14,14 +14,12 @@
 from cfme.infrastructure.provider import InfraProvider
 from cfme.infrastructure.virtual_machines import Vm, Template
 from cfme.web_ui import Quadicon, mixins, toolbar as tb
-from utils import providers
 from utils.appliance.implementations.ui import navigate_to
 
 
 @pytest.fixture(scope="module")
-def setup_first_provider():
-    providers.setup_a_provider_by_class(InfraProvider)
-    providers.setup_a_provider_by_class(CloudProvider)
+def cloud_and_infra_provider(cloud_provider, infra_provider):
```

Comment:
Unless we want to put "has_no_provider" everywhere, and that's still under the assumption that when we use e.g. `test_my_something(cloud_provider, has_no_provider):` we have a way to make sure that `has_no_providers` is executed before `cloud_provider`.